doc/book/app_server/luajit_memprof.rst

```diff
@@ -0,0 +1,524 @@
+            Overrides:
+                    @test.lua:0, line 13
+
+    @test.lua:0, line 14: 5 1984    992
+            Overrides:
+                    @test.lua:0, line 14
+
+
+    DEALLOCATIONS
+    INTERNAL: 20    0       1481
+    @test.lua:0, line 14: 3 0       7168
+            Overrides:
+                    @test.lua:0, line 14
+
+..  note::
+
+    On MacOS , a report is different because Tarantool and LuaJIT are built
+    with the GC64 mode enabled for this OS.
+
+Let's examine the report structure. A report has three sections:
+
+* ALLOCATIONS
+* RELOCATIONS
+* DEALLOCATIONS.
+
+Each section contains event records that are sorted from the most often
+to the least ones.
+
+An event record has the following format:
+
+..  code-block:: text
+
+    @<filename>:<function_line>, line <line_number>: <number_of_events> <allocated> <freed>
+
+*   <filename>—a name of the file containing Lua code.
+*   <function_line>—the line number where the function generating the event
+    is declared. In some of the cases, allocations are attributed not to
+    the declared function but to the main chunk. In this case, the <function_line>
+    is set to ``0``. See the :ref:`code chunk above<profiler_usage_example01>`
+    with the explanation in the comments for some examples.
+*   <line_number>—the line number where the event is detected.
+*   <number_of_events>—a number of events for this code line.
+*   <allocated>—amount of memory allocated during all the events, bytes.
+*   <freed>—amount of memory freed during all the events, bytes.
+
+The ``Overrides`` label shows what allocation has been overridden.
+
+.. _profiler_usage_internal_jitoff:
+
+The ``INTERNAL`` label indicates that this event is caused by internal LuaJIT
+structures.
+
+..  note::
+
+    Important note regarding the ``INTERNAL`` label and the recommendation
+    of switching the JIT compilation off (``jit.off()``): this version of the
+    profiler doesn't support verbose reporting for allocations on
+    `traces <https://en.wikipedia.org/wiki/Tracing_just-in-time_compilation#Technical_details>`_.
+    If memory allocations are made on a trace,
+    the profiler can't associate the allocations with the part of Lua code
+    that generated the trace. In this case, the profiler labels such allocations
+    as ``INTERNAL``.
+
+    So, if the JIT compilation is on,
+    new traces will be generated and there will be a mixture of events labeled
+    ``INTERNAL`` in the profiling report : some of them are really caused by
+    internal LuaJIT structures, but some of them are caused by allocations on
+    traces.
+
+    If you want to have more definite report without new trace allocations,
+    :ref:`call jit.off() <profiler_usage_example01>` before starting the profiling.
+    And if you want to completely exclude the trace allocations from the report,
+    remove also the old traces by additionally calling ``jit.flush()`` after
+    ``jit.off()``.
```

Comment:
Ough, this is a tough section even for me... Let's play with wording a bit here. In the first sentence you're talking about the allocations made while __generating__ a new trace, and they are internal by their design. In the second sentence you're talking about allocations made while __executing__ the generated trace, and they are internal due to the lack of debug machinery. So, I propose to use "JIT compiler allocations" for the first case and "trace allocations" for the second one. @Buristan, are you OK with it?

Reply:
Yes, I'm OK.

Reply:
Thanks, corrected.